packages/components/src/components/action-bar/action-bar.tsx

```diff
@@ -68,8 +71,33 @@ export class ActionBar extends LitElement {
     const groupCount =
       this.hasActionsEnd || !expandDisabled ? actionGroups.length + 1 : actionGroups.length;
 
+    let bufferSize = groupCount;
+    const actionBarContainerStyle = getComputedStyle(this.containerRef.value);
+
+    if (layout === "horizontal") {
+      bufferSize +=
+        parseInt(actionBarContainerStyle.paddingInlineStart) +
+        parseInt(actionBarContainerStyle.paddingInlineEnd);
+
+      if (this.actionGroups.length > 0) {
+        this.actionGroups.forEach((actionGroup, i) => {
+          const actionGroupStyle = getComputedStyle(actionGroup);
```

Comment:
Do you think we could assume the gap of one group is applicable to all? If so, it could help us reduce style computation.

Reply:
There could be user CSS that changes the gap for only some action groups, so this case wouldn't catch that.